Sources/MongoSwift/Cursor.swift

```diff
@@ -34,24 +34,50 @@ public class MongoCursor: Sequence, IteratorProtocol {
     }
 
     /**
-     * Returns the next document in this cursor, or nil
+     * Returns the next document in this cursor, or nil. Throws an error if one
+     * occurs. (Compared to next(), which returns nil and requires manually checking
+     * for an error afterward.)
+     *
      */
-    public func next() -> Document? {
+    func nextOrError() throws -> Document? {
```

Comment:
can't you reuse `next` in the body of this function instead of duplicating the code?

Reply:
yeah good point, can also use the `error` property to get the error, there was a lot of duplication here lol